test/end2end_test.go

```diff
@@ -6730,6 +6730,73 @@ func testRPCTimeout(t *testing.T, e env) {
 	}
 }
 
+type FakeCompressor struct {
+	name string
+}
+
+type customWriterCloser struct {
+	io.Writer
+}
+
+func (d *customWriterCloser) Close() error {
+	return nil
+}
+
+func (c *FakeCompressor) Compress(w io.Writer) (io.WriteCloser, error) {
+	customCloser := &customWriterCloser{w}
+	return customCloser, nil
+}
+
+func (c *FakeCompressor) Decompress(r io.Reader) (io.Reader, error) {
+	return bytes.NewBuffer(make([]byte, 10098)), nil
```

Comment:
This will decompress zeroes -- will it work with protobuf to deserialize this?

Reply:
I am not sure actually, i couldn't study the test server well enough. I went through a few examples used in tests so copied one of them, I'll admit that I don't see how decompression will work here. Will research more on this.